nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-shell-authorizer/src/main/java/org/apache/nifi/authorization/ShellUserGroupProvider.java

```diff
@@ -339,6 +347,26 @@ private long getDelayProperty(AuthorizerConfigurationContext authContext, String
         return syncInterval;
     }
 
+    private int getTimeoutProperty(AuthorizerConfigurationContext authContext, String propertyName, String defaultValue) {
+        final PropertyValue timeoutProperty = authContext.getProperty(propertyName);
+
+        final String propertyValue;
+        if (timeoutProperty.isSet()) {
+            propertyValue = timeoutProperty.getValue();
+        } else {
+            propertyValue = defaultValue;
+        }
+
+        final long timeoutValue;
+        try {
+            timeoutValue = Math.round(FormatUtils.getPreciseTimeDuration(propertyValue, TimeUnit.SECONDS));
+        } catch (final IllegalArgumentException ignored) {
+            throw new AuthorizerCreationException(String.format("The %s '%s' is not a valid time interval.", propertyName, propertyValue));
```

Comment:
Just thinking here -- should we throw an exception and stop the app from starting, or just warn loudly in the logs and fall back to the known good default value? If someone has manually configured this to be a different value, they probably had a good reason to do so, but then they should be expected to put it in a valid format...

Reply:
I could probably convinced of either side (lenient vs strict), but currently I think overall we tend to favor being strict. Looking through some other examples like LDAP and the existing Refresh Delay in ShellUserGroupProvider, those both fail start up for invalid formats, so I would probably leave as it is.

Reply:
Sounds good.